packages/esm-patient-common-lib/src/orders/useOrderableConceptSets.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  type FetchResponse,
+  openmrsFetch,
+  type OpenmrsResource,
+  reportError,
+  restBaseUrl,
+} from '@openmrs/esm-framework';
+import { useEffect, useMemo } from 'react';
+import useSWRImmutable from 'swr/immutable';
+
+export interface Concept {
+  uuid: string;
+  name: {
+    display: string;
+  };
+  names: Array<{
+    display: string;
+  }>;
+  conceptClass: {
+    uuid: string;
+  };
+  answers: Array<Concept>;
+  setMembers: Array<Concept>;
+  display: string;
+}
+
+type ConceptResult = FetchResponse<Concept>;
+type ConceptResults = FetchResponse<{ results: Array<Concept> }>;
+
+function openmrsFetchMultiple(urls: Array<string>) {
+  // SWR has an RFC for `useSWRList`:
+  // https://github.com/vercel/swr/discussions/1988
+  // If that ever is implemented we should switch to using that.
+  return Promise.all(urls.map((url) => openmrsFetch<{ results: Array<Concept> }>(url)));
+}
+
+function useOrderableConceptSWR(searchTerm: string, orderableConceptSets?: Array<string>) {
+  const { data, isLoading, error } = useSWRImmutable<Array<ConceptResult> | Array<ConceptResults>>(
+    orderableConceptSets?.length
+      ? orderableConceptSets.map(
+          (c) =>
+            `${restBaseUrl}/concept/${c}?v=custom:(display,names:(display),uuid,setMembers:(display,uuid,names:(display),setMembers:(display,uuid,names:(display))))`,
```

Comment:
Hi @ibacher!
The custom representation is calling for the `setMembers` for the fetched concept's setMembers.

Reply:
Right, so this will support this structure:

```
concept
    |
    ----> setMembers
                      |
                      ----> setMembers
```

But not this one:

```
concept
    |
    ----> setMembers
                      |
                      ----> setMembers
                                         |
                                         ----> setMembers
```

Or anything with deeper nesting. The problem is that this deeper nesting might be exactly what we need for other purposes.

I bring this up because our lab concepts, for example, are structured like:

```
Lab Orderables
          ↓
Hematology
          ↓
Bloodwork
```

This kind of structuring is useful for grouping orders into clinically useful sets.

Reply:
I'll implement it right away